mailbox/client.go

```diff
@@ -24,25 +31,40 @@ type Client struct {
 
 // NewClient creates a new Client object which will handle the mailbox
 // connection.
-func NewClient(ctx context.Context, connData *ConnData) (*Client, error) {
+func NewClient(ctx context.Context, serverHost string, connData *ConnData,
```

Comment:
Should we instead have two functions, `NewGrpcClient` and `NewWebsocketsClient` with the same signature?

Reply:
great idea!